tests/test_cli_compile.py

```diff
@@ -688,10 +643,10 @@ def test_annotate_option(pip_conf, runner, option, expected):
 
 
 @pytest.mark.parametrize(
-    "option, expected", [("--allow-unsafe", "\nsetuptools=="), (None, "\n# setuptools")]
+    "option, expected",
+    [("--allow-unsafe", "\nsmall-fake-a=="), (None, "\nsmall-fake-a")],
```

Comment:
```suggestion
    [("--allow-unsafe", "\nsmall-fake-a=="), (None, "\n# small-fake-a")],
```

The `#` symbol is important here. Without `--allow-unsafe` option an unsafe package (e.g. `setuptools`) becomes commented out. Since we removed `setuptools` dependency from `small-fake-with-deps` we need to make temporarily `small-fake-a` unsafe package for the test. I suggest to `monkeypatch` it by adding to the top of the test this:

```python
monkeypatch.setattr("piptools.resolver.UNSAFE_PACKAGES", {"small-fake-a"})
```

Reply:
Thanks for the suggestion. Done.